### DISCUSSION TITLE
Stuck in Linting and checking validity of types  ...

### DISCUSSION THREAD

**Discussion by smartdeveler** [vercel/next.js#74220] [App Router]
I add some feature to my project but When i want to build, Stuck in **"Linting and checking validity of types "**.
I use Material UI.

![Screenshot_20241222_103350](https://github.com/user-attachments/assets/9fb2d387-32f2-4258-a023-33d6795a1c0b)
.

**Comment by icyJoseph** [comment]
I once experienced this, and it was because a recursive TypeScript type definition that kept on expanding... do you have any complex types, or libraries that might use those?

**Reply by smartdeveler** [reply]
I don't think I have any recursive TypeScript type definitions.
